securityinternal/v1/types.go

```diff
@@ -0,0 +1,35 @@
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// RangeAllocation is used so we can easily expose a RangeAllocation typed for security group
+// This is an internal API, not intended for external consumption.
+type RangeAllocation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// range is a string representing a unique label for a range of uids, "1000000000-2000000000/10000".
+	Range string `json:"range"`
```

Comment:
define the regex pattern
Is this required?

Reply:
:+1: for regex and required.